contracts/coordinator/contracts/src/libs/LibEIP712Domain.sol

```diff
@@ -39,36 +49,70 @@ contract LibEIP712Domain {
+    // Hash of the EIP712 Domain Separator data for the Coordinator
+    // solhint-disable-next-line var-name-mixedcase
+    bytes32 public EIP712_COORDINATOR_DOMAIN_HASH;
+
+    // Hash of the EIP712 Domain Separator data for the Exchange
     // solhint-disable-next-line var-name-mixedcase
-    bytes32 public EIP712_DOMAIN_HASH;
+    bytes32 public EIP712_EXCHANGE_DOMAIN_HASH;
 
     constructor ()
         public
     {
-        EIP712_DOMAIN_HASH = keccak256(abi.encodePacked(
+        EIP712_COORDINATOR_DOMAIN_HASH = keccak256(abi.encodePacked(
             EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH,
-            keccak256(bytes(EIP712_DOMAIN_NAME)),
-            keccak256(bytes(EIP712_DOMAIN_VERSION)),
+            keccak256(bytes(EIP712_COORDINATOR_DOMAIN_NAME)),
+            keccak256(bytes(EIP712_COORDINATOR_DOMAIN_VERSION)),
             uint256(address(this))
         ));
+
+        EIP712_EXCHANGE_DOMAIN_HASH = keccak256(abi.encodePacked(
+            EIP712_DOMAIN_SEPARATOR_SCHEMA_HASH,
+            keccak256(bytes(EIP712_EXCHANGE_DOMAIN_NAME)),
+            keccak256(bytes(EIP712_EXCHANGE_DOMAIN_VERSION)),
+            uint256(address(EXCHANGE))
+        ));
     }
 
-    /// @dev Calculates EIP712 encoding for a hash struct in this EIP712 Domain.
+    /// @dev Calculates EIP712 encoding for a hash struct in the EIP712 domain
+    ///      of this contract.
     /// @param hashStruct The EIP712 hash struct.
     /// @return EIP712 hash applied to this EIP712 Domain.
-    function hashEIP712Message(bytes32 hashStruct)
+    function hashEIP712CoordinatorMessage(bytes32 hashStruct)
+        internal
+        view
+        returns (bytes32 result)
+    {
+        return hashEIP712Message(EIP712_COORDINATOR_DOMAIN_HASH, hashStruct);
+    }
+
+    /// @dev Calculates EIP712 encoding for a hash struct in the EIP712 domain
+    ///      of the Exchange contract.
+    /// @param hashStruct The EIP712 hash struct.
+    /// @return EIP712 hash applied to the Exchange EIP712 Domain.
+    function hashEIP712ExchangeMessage(bytes32 hashStruct)
         internal
         view
         returns (bytes32 result)
     {
-        bytes32 eip712DomainHash = EIP712_DOMAIN_HASH;
+        return hashEIP712Message(EIP712_EXCHANGE_DOMAIN_HASH, hashStruct);
+    }
 
+    /// @dev Calculates EIP712 encoding for a hash struct with a given domain hash.
+    /// @param eip712DomainHash Hash of the domain domain separator data.
+    /// @param hashStruct The EIP712 hash struct.
+    /// @return EIP712 hash applied to the Exchange EIP712 Domain.
+    function hashEIP712Message(bytes32 eip712DomainHash, bytes32 hashStruct)
+        internal
+        pure
+        returns (bytes32 result)
+    {
         // Assembly for more efficient computing:
         // keccak256(abi.encodePacked(
         //     EIP191_HEADER,
-        //     EIP712_DOMAIN_HASH,
-        //     hashStruct    
+        //     EIP712_COORDINATOR_DOMAIN_HASH,
```

Comment:
Should this be changed? I thought this function was agnostic to the domain specifics.